### PR TITLE
Print signal messages

### LINKED ISSUE #955: Segmentation faults and other special return codes should print corresponding messages
If an application like this:

```c
int main() {
    char* p = 0;
    return *p; // Uh-oh!
}
```

segfaults, Bash shows the following:

```
ryan@DevPC-LX:~/langtest/xonsh$ ./seg
Segmentation fault (core dumped)
ryan@DevPC-LX:~/langtest/xonsh$ 
```

But Xonsh shows:

```
ryan@DevPC-LX ~/langtest/xonsh $ ./seg

ryan@DevPC-LX ~/langtest/xonsh $
```

See:

http://tldp.org/LDP/abs/html/exitcodes.html

and:

https://en.wikipedia.org/wiki/Unix_signal#POSIX_signals

Would you be interested in a pull request for this?

### PR DESCRIPTION
**Pull request by kirbyfan64**
Closes #955.

Note that you should probably add a comment somewhere about how `wait_for_active_job` assigns to `obj.signal` if there was a signal sent. Took me ~30 minutes of debugging to figure out why `returncode` was always 0...

Also, I didn't see a place to add tests for this...

### CODE REVIEW

**Comment by scopatz** [comment]
Thanks for putting this in @kirbyfan64! It looks like it fails CI, though...

**Comment by kirbyfan64** [comment]
@scopatz Fixed!

File: xonsh/built_ins.py [hunk 1]
```diff
@@ -599,6 +610,16 @@ def run_subproc(cmds, captured=False):
                 errout = errout.replace('\r\n', '\n')
                 procinfo['stderr'] = errout
 
+    if getattr(prev_proc, 'signal', None):
+        sig, core = prev_proc.signal
+        try:
+            sig_str = SIGNAL_STRINGS[sig]
```
Comment (scopatz) [inline xonsh/built_ins.py]: Probably better to do a `SIGNAL_STRINGS.get()` here rather than waiting on an exception.

File: xonsh/built_ins.py [hunk 2]
```diff
@@ -41,6 +41,17 @@
 if ON_POSIX:
     AT_EXIT_SIGNALS += (signal.SIGTSTP, signal.SIGQUIT, signal.SIGHUP)
 
+SIGNAL_STRINGS = {
```
Comment (scopatz) [inline xonsh/built_ins.py]: Would you mind changing this to `SIGNAL_MESSAGES` or similar?

File: xonsh/built_ins.py [hunk 3]
```diff
@@ -599,6 +610,16 @@ def run_subproc(cmds, captured=False):
...
+    if getattr(prev_proc, 'signal', None):
+        sig, core = prev_proc.signal
+        try:
+            sig_str = SIGNAL_STRINGS[sig]
+        except KeyError:
+            pass
+        else:
+            if core:
+                sig_str += ' (core dumped)'
+            print(sig_str)
```
Comment (scopatz) [inline xonsh/built_ins.py]: This should probably be printed to stderr, rather than stdout.  Just do `print(sig_str, file=sys.stderr)`

**Comment by scopatz** [comment]
Thanks again @kirbyfan64, testing this might be kind of a pain... But you should at least put in a changelog entry :)

File: xonsh/built_ins.py (hunk 1 above)
Comment (kirbyfan64) [inline xonsh/built_ins.py]: Well, I was following the Python "easier to ask forgiveness than permission", but I'll use `get` if you prefer.

**Comment by kirbyfan64** [comment]
@scopatz Updated! Side note: I really like Xonsh's Git integration that shows the current branch!

**Comment by scopatz** [comment]
> @scopatz Updated! 

Thanks!  Could you add a changelog entry, though too?

> Side note: I really like Xonsh's Git integration that shows the current branch!

Awesome, thanks!  It works with hg too, for what that is worth :).

**Comment by kirbyfan64** [comment]
@scopatz Whoops, forgot that part! Done.

File: CHANGELOG.rst [hunk 4]
```diff
@@ -37,6 +37,8 @@ Current Developments
   ``--cache-everything`` when starting xonsh.
 * Added a workaround to allow ctrl-c to interrupt reverse incremental search in
   the readline shell
+* When a subprocess exits with a signal (e.g. SIGSEGV), a message is printed,
+  similar to Bash.
```
Comment (scopatz) [inline CHANGELOG.rst]: Haha whoops, can you merge from master and then move this bit to the new Current Developments section, please? Sorry for the run around, we did the v0.3.0 release in between when you last pulled or forked and now.

**Comment by kirbyfan64** [comment]
@scopatz Gah...done!

**Comment by scopatz** [comment]
Thanks!  Looks great to me!

**Comment by melund** [comment]
@scopatz @kirbyfan64 ... This PR broke master on Windows. I pushed an urgent fix #958

**Pull request merged**